core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -360,15 +359,17 @@ private[kafka] abstract class ConsumerStageLogic[K, V, Out](
   protected def poll(): Unit = {
     try {
       def toggleConsumption() = {
+        // note that consumer.assignment() is automatically changed when using
+        // dynamic subscriptions and we must use the current value to resume/pause
         if (isAvailable(out))
-          consumer.resume(assignments: _*)
+          consumer.resume(consumer.assignment().asScala.toArray: _*)
```

Comment:
yeah, that was why I optimized it previously (this was the original version), but that was before knowing how this actually worked.

Perhaps we can hand-roll the conversion to save the intermediate scala representation. In the end it's a varargs call. Stupid api.